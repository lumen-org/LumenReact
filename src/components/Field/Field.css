.field {
    border: solid 2px #0000;
    position: relative;
    margin: 0.2rem;
}

.is-hover {
    background: aqua;
}
