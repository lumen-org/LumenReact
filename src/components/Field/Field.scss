@import "src/assets/styles/colors";

.field {
    border: solid 2px $field-border-color;
    position: relative;
    margin: 0.2rem;
}

.is-hover {
    background: $field-hover-background;
}
