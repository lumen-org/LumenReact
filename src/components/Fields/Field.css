.field {
    border: solid 2px #0000;
    position: relative;
    margin: 0.2rem;
}
