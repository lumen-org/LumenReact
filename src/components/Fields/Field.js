import React from "react";
import TitelH2 from "../Titles/TitleH2";
import DataList from "../DataList";

function Field(props) {
  return (
      <div className="field">
        <TitelH2 value={props.title}/>
        <DataList value={props.data}/>
      </div>
  )
}

export default Field
