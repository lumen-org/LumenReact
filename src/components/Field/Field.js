import React  from "react";
import TitelH2 from "../Titles/TitleH2";
import DataList from "../DataList";
import "./Field.css";
import PropTypes from 'prop-types'

function Field({ title, data, fieldName="" }) {
  return (
    <div className={"Field"}>
      <TitelH2 value={title}/>
      <DataList dataList={data} fieldName={fieldName}/>
    </div>
  );
}

Field.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.array.isRequired
};

export default Field;
