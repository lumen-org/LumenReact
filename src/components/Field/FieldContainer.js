import React, { useState } from "react";
import { useDrop } from "react-dnd";
import classNames from "classnames";
import Field from "./Field";

function FieldContainer({ title, data, dropable = false }) {
  const [hasDropped, setHasDropped] = useState(false);
  const [{ isOverCurrent }, drop] = useDrop({
    accept: "DataItem",
    drop(item, monitor) {
      setHasDropped(true);
    },
    collect: monitor => ({
      isOverCurrent: monitor.isOver({ shallow: true })
    })
  });
  let fieldClasses = classNames({});
  if (dropable) {
    if (isOverCurrent) {
      fieldClasses += " is-hover";
    }
    return <div ref={drop} className={fieldClasses}>
      <Field title={title} data={data}/>
    </div>;
  }
  return (
    <div className={fieldClasses}>
      <Field title={title} data={data}/>
    </div>
  );
}

export default FieldContainer;
