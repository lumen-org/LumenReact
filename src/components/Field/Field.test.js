import React from "react";
import { unmountComponentAtNode } from "react-dom";
import TestRenderer from "react-test-renderer";
import Field from "./Field";

/// mock children components for shallow rendering
jest.mock("../../components/Titles/TitleH2", () => (props) => (
  <h2>{props.value}</h2>
));

jest.mock("../../components/FieldList/FieldList", () => (props) => (
<div>FieldList</div>
));

/// executed before every describe
let container = null;
let spy = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
  spy = jest.spyOn(console, 'error').mockImplementation(); /// prevents console from showing the errors while simultaniously being able to catch errors
});

/// executed after every describe
afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  console.clear();
  spy.mockRestore(); // important so errors are displayed again
});

/// Test missing Values
describe("Field.js Missing Values Test", () => {

  it("missing value data", () => {
    let title = "TestTitle";
    TestRenderer.create(<Field title={title} />);
    expect(console.error).toHaveBeenCalledTimes(1);
  });

  it("missing value title", () => {
    let data = ["TestTitle"];
    TestRenderer.create(<Field data={data} />);
    expect(console.error).toHaveBeenCalledTimes(1);
  }); 
});

/// test normal rendering and additional functionality
describe("Field.js Functionality Test", () => {
  it("render normally", () => {
    let title = "TestTitle";
    let data = ["Test", "Test", "Test"];
    const component = TestRenderer.create(<Field title={title} data={data} />);
    let tree = component.toJSON();
    expect(console.error).not.toBeCalled();
    expect(tree).toMatchSnapshot();
  });
});