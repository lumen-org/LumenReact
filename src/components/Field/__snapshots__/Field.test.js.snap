// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field.js Functionality Test render normally 1`] = `
<div
  className="Field"
>
  <h2>
    TestTitle
  </h2>
  <div>
    FieldList
  </div>
</div>
`;
